lib/widgets/create_post_popup_window.dart

```diff
@@ -1,120 +1,213 @@
+import 'dart:io';
 import 'package:flutter/material.dart';
-import 'package:flutter_svg/svg.dart';
-import 'package:twitter_clone/utils/colors.dart';
+import 'package:image_picker/image_picker.dart';
 
-class CreatePostPopupWindow extends StatelessWidget {
+class CreatePostPopupWindow extends StatefulWidget {
   final double screenHeight;
 
-  const CreatePostPopupWindow({super.key, required this.screenHeight});
+  const CreatePostPopupWindow({Key? key, required this.screenHeight}) : super(key: key);
+
+  @override
+  _CreatePostPopupWindowState createState() => _CreatePostPopupWindowState();
+}
+
+class _CreatePostPopupWindowState extends State<CreatePostPopupWindow> {
+  String? _pickedImagePath;
+  final ImagePicker _picker = ImagePicker();
+
+  // Callback function when selecting an image
+  void _pickImage() async {
```

Comment:
The pick image method should create in the `utils` folder for reuse in the future, in this case you can reuse in the create and update post screen